src/jvm/main/org/jetbrains/kotlinx/lincheck/transformation/LincheckClassVisitor.kt

```diff
@@ -91,6 +90,14 @@ internal class LincheckClassVisitor(
         }
         if (methodName == "<init>") {
             mv = ObjectCreationTransformer(fileName, className, methodName, mv.newAdapter())
+            mv = run {
```

Comment:
I did not try including `<init>` methods in the whole transformation chain. Right now I only require tracking fields/arrays accesses and other constructor invocations inside `<init>` method bodies. 
Maybe the feature of integrating constructors in full transformation chain should be a separate PR, since there are might be tricky spots where to account for https://github.com/JetBrains/lincheck/issues/424